x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/rerank/TextSimilarityRankBuilder.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.core.ml.rerank;
+
+import org.apache.lucene.search.Explanation;
+import org.apache.lucene.search.Query;
+import org.elasticsearch.TransportVersion;
+import org.elasticsearch.TransportVersions;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.search.rank.RankBuilder;
+import org.elasticsearch.search.rank.RankDoc;
+import org.elasticsearch.search.rank.context.QueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.QueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.context.RankFeaturePhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankCoordinatorContext;
+import org.elasticsearch.search.rank.rerank.RerankingQueryPhaseRankShardContext;
+import org.elasticsearch.search.rank.rerank.RerankingRankFeaturePhaseRankShardContext;
+import org.elasticsearch.xcontent.XContentBuilder;
+
+import java.io.IOException;
+import java.util.List;
+
+public class TextSimilarityRankBuilder extends RankBuilder {
+
+    private final String inferenceId;
+    private final String inferenceText;
+    private final String field;
+    private final float minScore;
+
+    public TextSimilarityRankBuilder(String field, String inferenceId, String inferenceText, int rankWindowSize, float minScore) {
+        super(rankWindowSize);
+        this.inferenceId = inferenceId;
+        this.inferenceText = inferenceText;
+        this.field = field;
+        this.minScore = minScore;
+    }
+
+    public TextSimilarityRankBuilder(StreamInput in) throws IOException {
+        super(in);
+        this.inferenceId = in.readString();
+        this.inferenceText = in.readString();
+        this.field = in.readString();
+        this.minScore = in.readFloat();
+    }
+
+    @Override
+    public String getWriteableName() {
+        return "text_similarity_rank_builder";
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.TEXT_SIMILARITY_RERANKER_RETRIEVER;
+    }
+
+    @Override
+    public void doWriteTo(StreamOutput out) throws IOException {
+        out.writeString(inferenceId);
+        out.writeString(inferenceText);
+        out.writeString(field);
+        out.writeFloat(minScore);
+    }
+
+    @Override
+    public void doXContent(XContentBuilder builder, Params params) throws IOException {
+        throw new UnsupportedOperationException("not supported");
```

Comment:
Is this a `TODO` or don't we need to implement this?

Reply:
Added implementation.